pyacq/core/sharedarray.py

```diff
@@ -55,13 +43,62 @@ def __init__(self, shape=(1,), dtype = 'float64', shm_id = None):
                 self.mmap = mmap.mmap(self._tmpFile.fileno(), self.nbytes, mmap.MAP_SHARED, mmap.PROT_READ)
                 
     def close(self):
+        """Close this buffer.
+        """
         self.mmap.close()
         if not sys.platform.startswith('win') and hasattr(self, '_tmpFile'):
             self._tmpFile.close()
     
     def to_dict(self):
-        return {'shape': self.shape, 'dtype': self.dtype, 'shm_id': self.shm_id}
+        """Return a dict that can be serialized and sent to other processes to
+        access this buffer.
+        """
+        return {'nbytes': self.nbytes, 'shm_id': self.shm_id}
+    
+    def to_numpy(self, offset, dtype, shape, strides):
+        """Return a numpy array pointing to part (or all) of this buffer.
+        """
+        return np.ndarray(buffer=self.mmap, shape=shape,
+                          strides=strides, offset=offset, dtype=dtype)        
+        
+
+
```

Comment:
For now, SharedArray is left in as a convenience, but I don't think it will end up being used. RingBuffer makes use of the lower-level SharedMem, which is only a 1-D byte buffer. This will allow more flexibility in how data is ordered when it is written into the buffer. In the end, I hope the user will never need to think about swapaxes--data will pass int a stream and arrive at the other end with the same shape and strides.